src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -51,6 +51,7 @@ def load_arguments(self, _):
                                                     help="a template file path in the file system")
     deployment_template_uri_type = CLIArgumentType(options_list=['--template-uri', '-u'], help='a uri to a remote template file')
     deployment_template_spec_type = CLIArgumentType(options_list=['--template-spec', '-s'], is_preview=True, min_api='2019-06-01', help="The template spec resource id.")
+    deployment_query_string_type = CLIArgumentType(options_list=['--query-string', '-q'], is_preview=True, help="The query string (a SAS token) to be used with the template-uri in the case of linked templates.")
```

Comment:
Does `--query-string` contain anything other than `sas-token`?
If `--query-string` is `sas-token`, why not name it `sas-token`?

There is `sas-token` parameter in storage module, so I personally think it's better to keep it consistent with storage
Such as:
```
storage file show -s {share} -p {file} --account-name {account} --sas-token
```

Reply:
It does not contain anything other than sas-token, it is named query string in PowerShell so it may be best to keep it consistent (https://github.com/Azure/azure-powershell/pull/13775).

Reply:
today it is only used for sasToken, but we'd like to leave open the possibility that something else could be passed besides a sas token. `--query-string` keeps it generic, while still being accurate.

Reply:
Got it, thanks~